Description: Allow SAGE_LOCAL to be overridden from outside the Makefiles
 This makes it easier for package managers to create binary distributions.
 However, keep SAGE_SPKG_INST relative to SAGE_ROOT rather than SAGE_LOCAL
 since that would be a more intrusive change.
Author: Ximin Luo <infinity0@debian.org>
Forwarded: TODO
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/sage/sage
+++ b/sage/sage
@@ -137,6 +137,8 @@
 elif [ -x "$SAGE_ROOT/local/bin/sage" ]; then # if in a stripped binary
     # Note in this case we assume that SAGE_LOCAL is the "local" subdirectory
     exec "$SAGE_ROOT/local/bin/sage" "$@"
+elif [ -x "$SAGE_LOCAL/bin/sage" ]; then # if in a stripped binary
+    exec "$SAGE_LOCAL/bin/sage" "$@"
 else
     echo >&2 "$0: no Sage installation found in \$SAGE_ROOT=$SAGE_ROOT"
     exit 1
--- a/sage/src/bin/sage-env
+++ b/sage/src/bin/sage-env
@@ -284,7 +284,7 @@
 export SAGE_ETC="$SAGE_LOCAL/etc"
 export SAGE_SHARE="$SAGE_LOCAL/share"
 export SAGE_PKGCONFIG="$SAGE_LOCAL/lib/pkgconfig"
-export SAGE_SPKG_INST="$SAGE_LOCAL/var/lib/sage/installed"
+export SAGE_SPKG_INST="$SAGE_ROOT/local/var/lib/sage/installed"
 export SAGE_SPKG_SCRIPTS="$SAGE_LOCAL/var/lib/sage/scripts"
 export SAGE_LOGS="$SAGE_ROOT/logs/pkgs"
 export SAGE_SRC="$SAGE_ROOT/src"
--- a/sage/src/sage/env.py
+++ b/sage/src/sage/env.py
@@ -164,11 +164,11 @@
 var('SAGE_INC',            join(SAGE_LOCAL, 'include'))
 var('SAGE_SHARE',          join(SAGE_LOCAL, 'share'))
 var('SAGE_DOC',            join(SAGE_SHARE, 'doc', 'sage'))
-var('SAGE_SPKG_INST',      join(SAGE_LOCAL, 'var', 'lib', 'sage', 'installed'))
 var('SAGE_LIB',            os.path.dirname(os.path.dirname(sage.__file__)))
 var('SAGE_EXTCODE',        join(SAGE_LIB, 'sage', 'ext_data'))
 
 var('SAGE_ROOT')           # no fallback for SAGE_ROOT
+var('SAGE_SPKG_INST',      join(SAGE_ROOT, 'local', 'var', 'lib', 'sage', 'installed'))
 var('SAGE_SRC',            join(SAGE_ROOT, 'src'), SAGE_LIB)
 var('SAGE_DOC_SRC',        join(SAGE_ROOT, 'src', 'doc'), SAGE_DOC)
 var('SAGE_PKGS',           join(SAGE_ROOT, 'build', 'pkgs'))
--- a/sage/build/make/install
+++ b/sage/build/make/install
@@ -18,7 +18,7 @@
 export SAGE_SHARE="$SAGE_LOCAL/share"
 export SAGE_PKGCONFIG="$SAGE_LOCAL/lib/pkgconfig"
 export SAGE_LOGS="$SAGE_ROOT/logs/pkgs"
-export SAGE_SPKG_INST="$SAGE_LOCAL/var/lib/sage/installed"
+export SAGE_SPKG_INST="$SAGE_ROOT/local/var/lib/sage/installed"
 . "$SAGE_SRC"/bin/sage-version.sh
 export SAGE_VERSION
 
@@ -28,7 +28,7 @@
 fi
 export PATH="$SAGE_ROOT/build/bin:$SAGE_SRC/bin:$SAGE_LOCAL/bin:$PATH"
 
-export PYTHONPATH="$SAGE_LOCAL"
+export PYTHONPATH="${PYTHONPATH:-SAGE_LOCAL}"
 
 ###############################################################################
 # Skip the rest if nothing to do (i.e., to [re]build).
--- a/sage/configure.ac
+++ b/sage/configure.ac
@@ -96,7 +96,7 @@
     SAGE_LOCAL=local
 fi
 SAGE_SRC="$SAGE_ROOT/src"
-SAGE_SPKG_INST="$SAGE_LOCAL/var/lib/sage/installed"
+SAGE_SPKG_INST="$SAGE_ROOT/local/var/lib/sage/installed"
 
 #---------------------------------------------------------
 
